### PR TITLE
Add GlobusDestination model

### PR DESCRIPTION
**Pull request by lwrubel**
# Why was this change made? 🤔
Refs #1188, to add a GlobusDestination model and a GlobusDestinations table to db.

# How was this change tested? 🤨
Unit

### CODE REVIEW

File: app/models/globus_destination.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+
+# Model representing a Globus destination that can be used as a staging location.
+class GlobusDestination < ApplicationRecord
+  belongs_to :batch_context, optional: true
```
Comment (edsu) [inline app/models/globus_destination.rb:5]: Is there value in relating the `GlobusDestination` and `User` models here? 

```ruby
belongs_to: user
```

That would give us model access for walking between them?

Comment (lwrubel) [reply app/models/globus_destination.rb:5]: I don't know. I noticed that a `BatchContext` belongs_to a `User`, so there is that relationship already. Would that help with what we need to connect up?

Comment (lwrubel) [reply app/models/globus_destination.rb:5]: Updated to add a foreign key for the `user` table row. It's possible to do a sunet lookup with something like `user.sunet_id` on the model, and as I do the next part with some helper methods on the model, will consider adding some a method for this if it seems useful.

**Comment by lwrubel** [comment]
I locally ran `bin/rails db:rollback` and that seemed to work smoothly. But if you see any gotchas I missed, let me know.

**Review by jmartin-sul — APPROVED** [review]

**Review by edsu — APPROVED** [review]

**Pull request merged**